sockeye/inference.py

```diff
@@ -691,12 +691,22 @@ def make_input_from_json_string(sentence_id: int, json_string: str) -> Translato
                 return _bad_input(sentence_id, reason=json_string)
         else:
             factors = None
+
+        # List of phrases that must appear in the output
         constraints = jobj.get(C.JSON_CONSTRAINTS_KEY)
         if isinstance(constraints, list) and len(constraints) > 0:
```

Comment:
Why the `isinstance`? Wouldn't it just work with `if constraints:`? Then you catch the case where it is `None` and len() == 0 in one go.

Reply:
Hmm, good point.